src/Storages/ObjectStorage/StorageObjectStorage.cpp

```diff
@@ -32,6 +33,19 @@ namespace ErrorCodes
     extern const int LOGICAL_ERROR;
 }
 
+
+bool checkIfHiveSettingEnabled(const ContextPtr & context, const std::string & storage_type_name)
+{
+    if (storage_type_name == "s3")
+        return context->getSettings().s3_hive_partitioning;
+    else if (storage_type_name == "hdfs")
+        return context->getSettings().hdfs_hive_partitioning;
+    else if (storage_type_name == "azure")
+        return context->getSettings().azure_blob_storage_hive_partitioning;
+    else
```

Comment:
Do we need a separate setting per storage? Maybe just one `use_hive_partitioning` is enough.

Also, it is not clear whether those settings are needed.
As I understand correctly, the only difference will be in the case when we specify a virtual column that is not in a table scheme. In this case, instead of throwing an exception, we can try to look for this column in hive partitioning.
(Or do we allow the mixed case, when some files have the virtual partition column but others don't?)

But we can have the setting if performance is affected.

Reply:
> But we can have the setting if performance is affected.

This is the reason. We should know all virtual columns during table creation, so it requires additional list request to get any path from globs when schema inference is not used (when it's used we will do list anyway and can reuse the result).